django/db/models/sql/query.py

```diff
@@ -400,15 +400,32 @@ def get_aggregation(self, using, aggregate_exprs):
         """
         if not aggregate_exprs:
             return {}
-        aggregates = {}
+        # Store annotation mask prior to temporarily adding aggregations
+        # for resolving purpose to facilitate their subsequent removal.
+        refs_subquery = False
+        replacements = {}
+        annotation_select_mask = self.annotation_select_mask
         for alias, aggregate_expr in aggregate_exprs.items():
             self.check_alias(alias)
             aggregate = aggregate_expr.resolve_expression(
                 self, allow_joins=True, reuse=None, summarize=True
             )
             if not aggregate.contains_aggregate:
                 raise TypeError("%s is not an aggregate expression" % alias)
-            aggregates[alias] = aggregate
+            # Temporarily add aggregate to annotations to allow remaining members
+            # of `aggregates` to resolve against each others.
+            self.append_annotation_mask([alias])
+            refs_subquery |= any(
+                getattr(self.annotations[ref], "subquery", False)
+                for ref in aggregate.get_refs()
+            )
+            aggregate = aggregate.replace_expressions(replacements)
+            self.annotations[alias] = aggregate
```

Comment:
I initially tried to add a `Query.summary : dict[str, Expression]` attribute that `resolve_refs` could dig into when trying to resolve references and avoid this whole add to annotations then pop shenanigans but that won't work for lookups because `build_filter` uses `names_to_path` instead (see ticket-24267's ultimate goal)